src/library/scala/collection/IterableOnce.scala

```diff
@@ -433,8 +433,8 @@ trait IterableOnceOps[+A, +CC[_], +C] extends Any { this: IterableOnce[A] =>
 
   def knownSize: Int = -1
 
-  @deprecated("Use .knownSize >=0 instead of .hasDefiniteSize", "2.13.0")
-  @`inline` final def hasDefiniteSize = knownSize >= 0
+  @deprecated("Use .knownSize >=0 instead of .hasDefiniteSize (note the slightly different semantics)", "2.13.0")
```

Comment:
Maybe describe these "slightly different semantics", or at least link to something that describes them ?

Reply:
I think there should actually be a scaladoc for it describing the semantics (and possibly also some of the limitations?), despite the fact that it's deprecated and to be removed.

Reply:
(for reference/copying, here's the scaladoc from 2.12)

```scala
  /** Tests whether this $coll is known to have a finite size.
    *  All strict collections are known to have finite size. For a non-strict
    *  collection such as `Stream`, the predicate returns `'''true'''` if all
    *  elements have been computed. It returns `'''false'''` if the stream is
    *  not yet evaluated to the end. Non-empty Iterators usually return
    *  `'''false'''` even if they were created from a collection with a known
    *  finite size.
    *
    *  Note: many collection methods will not work on collections of infinite sizes.
    *  The typical failure mode is an infinite loop. These methods always attempt a
    *  traversal without checking first that `hasDefiniteSize` returns `'''true'''`.
    *  However, checking `hasDefiniteSize` can provide an assurance that size is
    *  well-defined and non-termination is not a concern.
    *
    *  @return  `'''true'''` if this collection is known to have finite size,
    *           `'''false'''` otherwise.
    */
```